recipes/agent.rb

```diff
@@ -1,52 +1,41 @@
 include_recipe "zabbix::agent_#{node['zabbix']['agent']['install_method']}"
 include_recipe 'zabbix::agent_common'
 
-# Install configuration
-template 'zabbix_agentd.conf' do
-  path node['zabbix']['agent']['config_file']
-  source 'zabbix_agentd.conf.erb'
-  unless node['platform_family'] == 'windows'
+# Install configuration and optional additional agent config file containing UserParameter(s)
+case node['platform_family']
+when 'windows'
+  template 'zabbix_agentd.conf' do
```

Comment:
I noticed that none of the variables that are used in the template here are actually described in the resource. Ideally we [shouldn't be using attributes in Chef templates at all](https://discourse.chef.io/t/attributes-in-templates/8493). We should pass them in as [variables](https://docs.chef.io/resource_template.html#variables) to the `template` resource.

I'll let this slide since the rest of this cookbook's templates don't work that way, but we should pay attention to not do this in the future.